cleancode/src/main/java/com/epam/cleancode/comments/MortgageInstallmentCalculator.java

```diff
@@ -2,38 +2,42 @@
 
 public class MortgageInstallmentCalculator {
 
+    private static final String INVALID_INPUT_EXCEPTION = "Negative values are not allowed";
+    private static final double NUMBER_OF_MONTHS = 12.0;
+
     /**
      *
-     * @param p principal amount
-     * @param t term of mortgage in years
-     * @param r rate of interest
-     * @return monthly payment amount
+     * @param principalAmount principal amount
+     * @param termOfMortgageInYears term of mortgage in years
+     * @param rateOfInterest rate of interest
+     * @return  monthly payment amount
      */
-    public static double calculateMonthlyPayment(
-            int p, int t, double r) {
-
-        //cannot have negative loanAmount, term duration and rate of interest
-        if (p < 0 || t <= 0 || r < 0) {
-            throw new InvalidInputException("Negative values are not allowed");
-        }
 
+    static double calculateMonthlyPayment(int principalAmount, int termOfMortgageInYears, double rateOfInterest) {
         // Convert interest rate into a decimal - eg. 6.5% = 0.065
-        r /= 100.0;
+        rateOfInterest /= 100.0;
 
         // convert term in years to term in months
-        double tim = t * 12;
+        double termInMonths = termOfMortgageInYears * NUMBER_OF_MONTHS ;
 
         //for zero interest rates
-        if(r==0)
-            return  p/tim;
+        if(rateOfInterest==0)
+            return  principalAmount/termInMonths;
 
         // convert into monthly rate
-        double m = r / 12.0;
-
+        double monthlyRate = rateOfInterest / NUMBER_OF_MONTHS;
         // Calculate the monthly payment
         // The Math.pow() method is used calculate values raised to a power
-        double monthlyPayment = (p * m) / (1 - Math.pow(1 + m, -tim));
+        double monthlyPayment = (principalAmount * monthlyRate) / (1 - Math.pow(1 + monthlyRate, -termInMonths));
+        checkValid(principalAmount, termOfMortgageInYears, rateOfInterest);
```

Comment:
Should be on top of the function, because you currently make some unnecessary calculation in case invalid parameters